File: src/intellibitz/sted/actions/TableRowsSelectAction.java
package intellibitz.sted.actions;

import intellibitz.sted.fontmap.FontMapEntry;
import intellibitz.sted.ui.MappingTableModel;

import javax.swing.*;
import javax.swing.event.ListSelectionListener;
import javax.swing.table.TableModel;
import java.util.ArrayList;
import java.util.Collection;

abstract public class TableRowsSelectAction
        extends STEDWindowAction
        implements ListSelectionListener {
    private JTable table;

    protected TableRowsSelectAction() {
        super();
    }

    TableModel getTableModel() {
        return table.getModel();
    }

    public void setTable(JTable table) {
        this.table = table;
    }

    void selectAll() {
        table.selectAll();
    }

    Collection<FontMapEntry> getSelectedRows() {
        final int row[] = table.getSelectedRows();
        final Collection<FontMapEntry> rows =
                new ArrayList<FontMapEntry>(row.length);
        for (final int newVar : row) {
            rows.add(((MappingTableModel) table.getModel()).getValueAt(newVar));
        }
        return rows;
    }

    Collection copySelectedRows() {
        final int row[] = table.getSelectedRows();
        final Collection<FontMapEntry> rows =
                new ArrayList<FontMapEntry>(row.length);
        for (final int newVar : row) {
            rows.add((FontMapEntry) ((MappingTableModel) table.getModel())
                    .getValueAt(newVar).clone());
        }
        return rows;
    }
}

File: src/main/intellibitz/sted/actions/TableRowsSelectAction.kt
package sted.actions

import sted.fontmap.FontMapEntry
import sted.ui.MappingTableModel
import java.util.*
import javax.swing.JTable
import javax.swing.event.ListSelectionListener

abstract class TableRowsSelectAction protected constructor() : STEDWindowAction(), ListSelectionListener {
    lateinit var table: JTable
    val tableModel: MappingTableModel
        get() = table.model as MappingTableModel

    fun selectAll() {
        table.selectAll()
    }

    val selectedRows: Collection<FontMapEntry>
        get() {
            val row = table.selectedRows
            val rows: MutableCollection<FontMapEntry> = ArrayList(row.size)
            for (newVar in row) {
                rows.add(tableModel.getValueAt(newVar)!!)
            }
            return rows
        }

    fun copySelectedRows(): Collection<*> {
        val row = table.selectedRows
        val rows: MutableCollection<FontMapEntry> = ArrayList(row.size)
        for (newVar in row) {
            rows.add(
                tableModel.getValueAt(newVar)?.clone() as FontMapEntry
            )
        }
        return rows
    }
}